File: frontend/apps/shell/src/api/auditKhktThang.ts
import type { ApiResponse } from "@govia/ui-kit";
import { httpClient } from "./client";

export interface AuditKhktThangRowItem {
  year: number;
  auditObjectCode: string;
  auditObjectName: string;
  auditObjectCategoryCode: string | null;
  rankLabel: string | null;
  onBalanceSheetLoan: number | null;
  fundingSource: number | null;
  creditScale: number | null;
  fundingScale: number | null;
  geographicArea: string | null;
  businessSegmentCodes: string[];
  month1: boolean;
  month2: boolean;
  month3: boolean;
  month4: boolean;
  month5: boolean;
  month6: boolean;
  month7: boolean;
  month8: boolean;
  month9: boolean;
  month10: boolean;
  month11: boolean;
  month12: boolean;
  note: string | null;
}

export interface AuditKhktThangUpdateRequest {
  month1: boolean;
  month2: boolean;
  month3: boolean;
  month4: boolean;
  month5: boolean;
  month6: boolean;
  month7: boolean;
  month8: boolean;
  month9: boolean;
  month10: boolean;
  month11: boolean;
  month12: boolean;
  note: string | null;
}

const BASE = "/api/audit/plan/khkt-thang";

export async function listAuditKhktThang(year: number): Promise<AuditKhktThangRowItem[]> {
  const res = await httpClient.get<ApiResponse<AuditKhktThangRowItem[]>>(BASE, { params: { year } });
  return res.data.data;
}

export async function updateAuditKhktThang(
  year: number,
  auditObjectCode: string,
  request: AuditKhktThangUpdateRequest,
): Promise<AuditKhktThangRowItem> {
  const res = await httpClient.put<ApiResponse<AuditKhktThangRowItem>>(
    `${BASE}/${year}/${encodeURIComponent(auditObjectCode)}`,
    request,
  );
  return res.data.data;
}

File: backend/govia-identity/src/main/java/com/govia/audit/khkt/thang/entity/AuditKhktThang.java
package com.govia.audit.khkt.thang.entity;

import com.govia.core.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

/** "Khai bao so thang kiem toan trong nam" (sheet ZTC_KHKT_THANG, bang ZTB_KHKT_THANG) - phan tich
 * chon (kiem toan hay khong) cho tung thang trong nam, cho 1 doi tuong kiem toan da co trong TH2
 * (AuditKhktThConfirmed) cua nam do. Cac cot con lai cua man hinh (ten, phan loai, xep hang, du no,
 * nguon von, khu vuc dia ly, linh vuc kiem toan, quy mo) deu doc song tu TH2/AuditObjectUnit/
 * AuditKhktScale - bang nay CHI luu phan NSD tu nhap (12 thang + ghi chu). */
@Getter
@Setter
@Entity
@Table(name = "audit_khkt_thang")
public class AuditKhktThang extends BaseEntity {

    @Column(name = "khkt_year", nullable = false)
    private Integer year;

    @Column(name = "audit_object_code", nullable = false, length = 20)
    private String auditObjectCode;

    @Column(name = "month1", nullable = false)
    private boolean month1 = false;

    @Column(name = "month2", nullable = false)
    private boolean month2 = false;

    @Column(name = "month3", nullable = false)
    private boolean month3 = false;

    @Column(name = "month4", nullable = false)
    private boolean month4 = false;

    @Column(name = "month5", nullable = false)
    private boolean month5 = false;

    @Column(name = "month6", nullable = false)
    private boolean month6 = false;

    @Column(name = "month7", nullable = false)
    private boolean month7 = false;

    @Column(name = "month8", nullable = false)
    private boolean month8 = false;

    @Column(name = "month9", nullable = false)
    private boolean month9 = false;

    @Column(name = "month10", nullable = false)
    private boolean month10 = false;

    @Column(name = "month11", nullable = false)
    private boolean month11 = false;

    @Column(name = "month12", nullable = false)
    private boolean month12 = false;

    @Column(name = "note", length = 250)
    private String note;
}
